state/address.go

```diff
@@ -310,23 +310,36 @@ func (st *State) apiHostPortsForCAAS(public bool) (addresses []network.SpaceHost
 		return addrsToHostPorts(publicAddrs...), nil
 	}
 
+	var hostAddresses network.SpaceAddresses
+	// Add in the FQDN of the controller service for agents to use as an option.
+	controllerName := controllerConfig.ControllerName()
+	if controllerName != "" {
+		hostAddresses = append(
+			hostAddresses, network.NewScopedSpaceAddress(
+				fmt.Sprintf("controller-service.controller-%s.svc.cluster.local", controllerName),
```

Comment:
@wallyworld this should probably be in github.com/juju/juju/caas/kubernetes/provider/constants

Reply:
Also maybe a comment saying this is a caas todo hack